### ISSUE TITLE
KeyError after deleting an .rst file

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#573]
(This is a different error than <<issue 515>>.) I deleted a file "api.rst" and the ".. toctree" that referenced it. The next day I ran "make html" and got:

```
writing output... [ 50&#37;] api                    
Exception occurred:
  File "/usr/lib/pymodules/python2.6/sphinx/environment.py", line 931, in get_toc_for
    toc = self.tocs[docname].deepcopy()
KeyError: 'api'
The full traceback has been saved in /tmp/sphinx-err-5oxkBE.log, if you want to report the issue to the developers.                                             
Please also report this if it was a user error, so that a better error message can be provided next time.
Either send bugs to the mailing list at <http://groups.google.com/group/sphinx-dev/>,
or report them in the tracker at <https://bitbucket.org/birkenfeld/sphinx/issues/>. Thanks!
```

I deleted the "_build" directory and the error disappeared. So presumably it's not recognizing that the state pickle is out of date.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/573
- Originally reported by: Mike Orr
- Originally created at: 2010-12-23T22:55:41.682

**Comment by shimizukawa** [comment]
_From Mike Orr on 2010-12-23 21:57:21+00:00_

Full stack trace uploaded as "sphinx-err-50xxkBE.log".

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-01-03 18:57:20+00:00_

I know that error, I've come across it once or twice, intending to debug it.

However, now I can't reproduce it.  I'll keep it in mind and really debug it when it happens to me again.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-01-07 14:03:38+00:00_

Should now be fixed in <<cset 6c36366cc1f2581f86eef274f6cbfb47aa6ad9b8>>.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-10-18 07:30:27+00:00_

Removing version: 1.0.4 (automated comment)
